### PR TITLE
Add support for Time and Time64 types

### PR DESCRIPTION
**Pull request by nohupped**
## Summary
Add support for Time and Time64 types
Add Integration test for both types using Time and Chrono
Updated Readme with examples

Reference: https://github.com/ClickHouse/clickhouse-rs/issues/242

related to https://github.com/ClickHouse/ClickHouse/pull/81217
## Checklist
Delete items not relevant to your PR:
- [x] Unit and integration tests covering the common scenarios were added
- [x] A human-readable description of the changes was provided so that we can include it in CHANGELOG later
- [x] For significant changes, documentation in README and https://github.com/ClickHouse/clickhouse-docs was updated with further explanations or tutorials

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -400,6 +400,38 @@ How to choose between all these features? Here are some considerations:
...
+    <details>
+    <summary>Example</summary>
+
+    ```rust,ignore
+    #[derive(Row, Serialize, Deserialize)]
+    struct MyRow {
+        #[serde(with = "clickhouse::serde::time::time")]
+        t0: Time,
+    }
+
```
Comment (slvrtrn) [inline README.md:434]: CH supports Time range in `[-999:59:59, 999:59:59]`, with up to nanos precision in Time64. 

Perhaps it's better to map it to Duration instead? `chrono::Duration` supports negative values; but `std::time::Duration` does not seem to though...

File: examples/time_types_example.rs [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        .fetch_all()
+        .await?;
+    for time_example in rows {
+        println!("Time example: {time_example:?}");
+    }
+
+    println!("✅ Time and Time64 types example completed successfully!");
+
+    Ok(())
+}
```
Comment (slvrtrn) [inline examples/time_types_example.rs:131]: We will need an example with corner cases too, I think. Something like `-999:59:59`, `0`, `999:59:59`

File: src/rowbinary/tests.rs [hunk 3]
```diff
@@ -7,6 +7,12 @@ struct Timestamp32(u32);
 #[derive(Debug, PartialEq, Serialize, Deserialize)]
 struct Timestamp64(u64);
 
+#[derive(Debug, PartialEq, Serialize, Deserialize)]
+struct Time32(u32);
+
+#[derive(Debug, PartialEq, Serialize, Deserialize)]
+struct Time64(u64);
```
Comment (slvrtrn) [inline src/rowbinary/tests.rs]: Is it signed or unsigned? Cause in the README, it is mentioned as a signed integer for both.

File: types/src/data_types.rs [hunk 4]
```diff
@@ -1364,6 +1465,50 @@ mod tests {
...
+            DataTypeNode::new("Time('UTC')").unwrap(),
+            DataTypeNode::Time
+        );
+        assert_eq!(
+            DataTypeNode::new("Time('Europe/Moscow')").unwrap(),
+            DataTypeNode::Time
+        );
+        // Edge cases for Time64
+        use super::DateTimePrecision::*;
+        // Midnight
```
Comment (slvrtrn) [inline types/src/data_types.rs]: Midnight?

File: types/src/data_types.rs [hunk 5]
```diff
@@ -1364,6 +1465,50 @@ mod tests {
...
+        assert_eq!(
+            DataTypeNode::new("Time64(0, 'UTC')").unwrap(),
+            DataTypeNode::Time64(Precision0)
+        );
+        // Random value (precision 3, no tz)
+        assert_eq!(
+            DataTypeNode::new("Time64(3)").unwrap(),
+            DataTypeNode::Time64(Precision3)
+        );
+        // Random value (precision 6, with tz)
```
Comment (slvrtrn) [inline types/src/data_types.rs]: Most of the comments seem related to another test here

File: README.md (hunk 1 above)
Comment (slvrtrn) [reply README.md:434]: CC @serprex @loyd PTAL

Comment (nohupped) [reply README.md:434]: Is it acceptable to use chrono::Duration for the chrono implementation, while use time::Time in the other?

Comment (nohupped) [reply README.md:434]: I've refactored the NaiveTime to Duration, while keeping the `time::Time` as it is, as it supports negative values.
Please let me know what you think. Thanks

File: examples/time_types_example.rs (hunk 2 above)
Comment (nohupped) [reply examples/time_types_example.rs:131]: Added.

File: types/src/data_types.rs (hunk 4 above)
Comment (nohupped) [reply types/src/data_types.rs]: Cleaned up old comments.

File: types/src/data_types.rs (hunk 5 above)
Comment (nohupped) [reply types/src/data_types.rs]: Cleaned up old comments

File: README.md (hunk 1 above)
Comment (nohupped) [reply README.md:434]: I read the documentation about time::Duration. It does support negative values. I've refactored the serde for Time feature to use `time::Duration` instead of `time::Time` and added test cases for both.
```
SELECT *
FROM chrs__time__time64_negative_roundtrip.test_time64_negative

Query id: 83162fa4-8abd-48c6-8427-9def9f072a00

   ┌────────t0─┬────t0_opt─┬────────────t3─┬───────────────t6─┬──────────────────t9─┐
1. │ -05:15:30 │ -05:15:30 │ -05:15:30.123 │ -05:15:30.123456 │ -05:15:30.123456789 │
   └───────────┴───────────┴───────────────┴──────────────────┴─────────────────────┘

1 row in set. Elapsed: 0.002 sec.

SELECT *
FROM chrs__chrono__time64_negative_roundtrip.test_time64_negative

Query id: 3df53891-93f0-4954-86aa-40979686e3d6

   ┌────────t0─┬────t0_opt─┬────────────t3─┬────────t3_opt─┬───────────────t6─┬───────────t6_opt─┬──────────────────t9─┬──────────────t9_opt─┐
1. │ -05:15:30 │ -05:15:30 │ -05:15:30.123 │ -05:15:30.123 │ -05:15:30.123456 │ -05:15:30.123456 │ -05:15:30.123456789 │ -05:15:30.123456789 │
   └───────────┴───────────┴───────────────┴───────────────┴──────────────────┴──────────────────┴─────────────────────┴─────────────────────┘

1 row in set. Elapsed: 0.020 sec.

Comment (serprex) [reply README.md:434]: +1 on duration, ended up coming to the same conclusion in PeerDB dealing with MySQL's time type (which is very similar to ClickHouse's time type) in https://github.com/PeerDB-io/peerdb/pull/3036

**Review by serprex — DISMISSED** [review]

File: README.md [hunk 6]
```diff
@@ -400,6 +400,38 @@ How to choose between all these features? Here are some considerations:
     }
 
 
+    ```
+    </details>
+* `Time` maps to/from i32 or a newtype around it and represents the time‐of‐day value on its own, without any date or regional offset context. Attempting to apply or change a time zone on Time columns has no effect and is not supported.
```
Comment (slvrtrn) [inline README.md]: It's not a time-of-day, as it is modelled after MySQL's time, which is kinda "duration" or "interval" type: https://dev.mysql.com/doc/refman/8.4/en/time.html. See also: https://clickhouse.com/docs/sql-reference/data-types/time; we can just copy:

> The Time data type is used to store a time value independent of any calendar date. It is ideal for representing daily schedules, event times, or any situation where only the time component (hours, minutes, seconds) is important.

File: README.md [hunk 7]
```diff
@@ -400,6 +400,38 @@ How to choose between all these features? Here are some considerations:
...
+    struct MyRow {
+        #[serde(with = "clickhouse::serde::chrono::time64::secs")]
+        t0: chrono::Duration,
+        #[serde(with = "clickhouse::serde::chrono::time64::secs::option")]
+        t0_opt: Option<chrono::Duration>,
+    }
+
+    ```
+    </details>
+* `Time64(_)` maps to/from i64 or a newtype around it and represents the time‐of‐day with sub-second precision. Unlike DateTime64, Time64 does not store a date component, meaning that it only represents time.
```
Comment (slvrtrn) [inline README.md]: Similar to Time, it is not a time-of-day value.

File: types/src/data_types.rs [hunk 8]
```diff
@@ -132,10 +137,13 @@ impl DataTypeNode {
             "MultiLineString" => Ok(Self::MultiLineString),
             "Polygon" => Ok(Self::Polygon),
             "MultiPolygon" => Ok(Self::MultiPolygon),
+            "Time" => Ok(Self::Time),
```
Comment (slvrtrn) [inline types/src/data_types.rs]: I think you can remove this branch and leave just:

```rs
str if str.starts_with("Time64") => parse_time64(str),
str if str.starts_with("Time") => Ok(Self::Time), // ignore timezone
```

below

**Review by slvrtrn — DISMISSED** [review]
Let's fix README and merge

**Review by serprex — APPROVED** [review]
